src/components/Divider.tsx

```diff
@@ -9,27 +9,36 @@ import head from '../util/head'
 
 /** A custom horizontal rule. */
 const Divider = ({ path }: { path: Path }) => {
-  const dividerSetWidth = React.createRef<HTMLInputElement>()
+  const dividerRef = React.createRef<HTMLInputElement>()
   const dispatch = useDispatch()
+  const [width, setWidth] = useState(DIVIDER_MIN_WIDTH)
 
   /** Sets the cursor to the divider. */
   const setCursorToDivider = (e: React.MouseEvent | React.TouchEvent) => {
     e.stopPropagation()
     dispatch(setCursor({ path }))
   }
 
-  /** Get the max width of nearby for divider list child elements, add 30 px and set this width for divider. */
+  /** Get the max width of nearby for divider list child elements, add DIVIDER_PLUS_PX and set this width for divider. */
   const setStyle = () => {
-    if (dividerSetWidth.current) {
-      const parentUl = dividerSetWidth.current.closest('ul')
-      const children = parentUl ? (Array.from(parentUl.childNodes) as HTMLElement[]) : []
-      const widths = children.map((child: HTMLElement) => {
-        if (child.classList.contains('child-divider')) return DIVIDER_PLUS_PX
-        const subs = child.getElementsByClassName('subthought') as HTMLCollectionOf<HTMLElement>
-        return subs.length ? subs[0].offsetWidth + DIVIDER_PLUS_PX : DIVIDER_PLUS_PX
-      })
-      const maxWidth = Math.max(...widths)
-      dividerSetWidth.current.style.width = `${maxWidth > DIVIDER_MIN_WIDTH ? maxWidth : DIVIDER_MIN_WIDTH}px`
+    if (dividerRef.current) {
+      const parentNode = dividerRef.current.closest('div.tree-node')
+
+      /** Get the depth of this thought's node. */
+      const parentDepth = parentNode ? parseInt(parentNode.getAttribute('data-depth') || '0') : 0
+      const siblingNodes = Array.from(document.querySelectorAll(`.tree-node[data-depth="${parentDepth}"]`))
+
+      /** Find the largest child node and set the width of the divider to its width plus DIVIDER_PLUS_PX. */
+      const maxWidth = Math.max(
+        ...siblingNodes
+          .filter(node => !node.querySelector('.divider'))
```

Comment:
Because this function will be called on every keystroke, I'd like to do a small performance optimization.

Instead of filtering the nodes here, let's add a `thought-divider` class to [StaticThought](https://github.com/cybersemics/em/blob/d1781ab3fc3db39a64ab15caba2e8625a5c3bead/src/components/StaticThought.tsx#L160) when it is a divider thought. Then you can exclude the divider in the `.map` below without an additional DOM query.